### PR TITLE
Features/aaronjeline/timebox

### PR DESCRIPTION
**Pull request by aaronjeline**
Adds timeboxed read-only shares.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tinytodo/src/api.rs [hunk 1]
```diff
@@ -71,6 +71,29 @@ pub struct AddShare {
...
     pub share_with: UserOrTeamUid,
     pub role: ShareRole,
+    pub duration_in_seconds: Option<u64>,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum ShareKind {
+    Read,
+    Edit,
+    Timebox,
```
Comment (mwhicks1) [inline tinytodo/src/api.rs:81]: I'd rename this to `TimeboxRead` so its meaning is clear in isolation.

File: tinytodo/src/api.rs [hunk 2]
```diff
@@ -93,6 +116,15 @@ pub struct DeleteShare {
     pub role: ShareRole,
 }
 
+impl DeleteShare {
+    pub fn share_kind(&self) -> ShareKind {
+        match self.role {
+            ShareRole::Reader => ShareKind::Read,
```
Comment (mwhicks1) [inline tinytodo/src/api.rs:122]: We are not allowed to delete timeboxed shares?

File: tinytodo/src/context.rs [hunk 3]
```diff
@@ -288,24 +282,83 @@ impl AppContext {
...
+                r.share_with,
+                r.list,
+                r.duration_in_seconds.map(Duration::from_secs),
+            )?;
+        }
+
         Ok(AppResponse::Unit(()))
     }
 
+    #[tracing::instrument]
```
Comment (mwhicks1) [inline tinytodo/src/context.rs:309]: All of the functionality to manage and update timeboxes and such doesn't feel like it belongs in `context.rs`, which ought to just have the handlers for the commands. Can you move it somewhere else, e.g., to `util.rs` ?

Comment (mwhicks1) [reply tinytodo/src/context.rs:309]: Actually, I'd make a new file, `timebox.rs` and put all the timebox related stuff there, other than the outermost handler.

**Review by mwhicks1 — COMMENTED** [review]
Main comment: Move timebox functionality to its own file, rather than have a bunch in the handlers file, a bunch in the objects file, etc. I suspect it will be easier to understand by collecting it together, and the diff with the main branch will also be easier.

**Review by mwhicks1 — APPROVED** [review]
Looks great!

But: You need to update `TUTORIAL.md` still, right? I guess that's on the main branch (and then rolled forward to this one)

File: tinytodo/src/timebox.rs [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+use crate::{
+    context,
+    entitystore::EntityStore,
+    objects::Timebox,
+    policy_store::PolicyStore,
+    util::{EntityUid, ListUid, TimeBoxUid, UserOrTeamUid, TYPE_TIMEBOX, TYPE_USER},
+};
+
+#[tracing::instrument]
+pub fn update_timebox(
```
Comment (andrewmwells-amazon) [inline tinytodo/src/timebox.rs:17]: [nit] It might be better to define a struct to pass in as the argument instead of passing in 5 arguments.

**Review by andrewmwells-amazon — APPROVED** [review]

**Review by andrewmwells-amazon — APPROVED** [review]

**Comment by aaronjeline** [comment]
Closing as we don't want to merge into `main`

**Pull request closed without merging**